language-support/ts/daml-ledger/index.ts

```diff
@@ -953,6 +955,61 @@ class Ledger {
     return decode(partyInfoDecoder, json);
   }
 
+  /**
+   * Fetch a list of all package IDs from the ledger.
+   *
+   * @returns List of package IDs.
+   *
+   */
+  async listPackages(): Promise<PackageId[]> {
+    const json = await this.submit('v1/packages', undefined, 'get');
+    return decode(jtv.array(jtv.string()), json);
+  }
+
+  /**
+   * Fetch a binary package.
+   *
+   * @returns The content of the package as a [Blob], or a list of errors.
+   *
+   * [Blob]: https://developer.mozilla.org/en-US/docs/Web/API/Blob
+   *
+   */
+  async getPackage(id: PackageId): Promise<{success: true; package: ArrayBuffer } | {success: false; errors: string[]}> {
```

Comment:
The types here seem a bit odd. Why are `getPackage` and `uploadDarFile` the only things where we return errors as something like an `Either` instead of failing the promise?

Reply:
Good catch, I guess I never paid much attention to failing cases so far. I don't think we have any test for them. I'll switch to exception-based API for consistency.